spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -842,11 +853,24 @@ private static boolean hasReturnParameterizedWithMessage(Method method, boolean
 		return false;
 	}
 
+	private boolean hasReturnMessageTypeOnFunction(Method method) {
+		if (Function.class.isAssignableFrom(this.serviceInterface) && "apply".equals(method.getName())) {
+			Class<?> returnType =
+					ResolvableType.forClass(Function.class, this.serviceInterface)
+							.getGeneric(1)
+							.getRawClass();
+			return returnType != null && Message.class.isAssignableFrom(returnType);
```

Comment:
Again, how would it be possible to have `returnType` null on the Function.apply()?

Reply:
This is syntax sugar to make our IDE and Sonar happy.
this is just because `ResolvableType.getRawClass()` is `@Nullable`, so sending `null` as argument into `isAssignableFrom()` is bad practice.